tests/integration/e2e/test_password_rotation.py

```diff
@@ -0,0 +1,112 @@
+#!/usr/bin/env python3
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+import asyncio
+import logging
+
+import pytest
+from literals import DATABASE_CHARM_NAME, KAFKA_CHARM_NAME, ZOOKEEPER_CHARM_NAME
+from pytest_operator.plugin import OpsTest
+from tests.integration.e2e.helpers import (
+    check_produced_and_consumed_messages,
+    fetch_action_get_credentials,
+)
+
+logger = logging.getLogger(__name__)
+
+TOPIC = "topic_1"
```

Comment:
**todo** can we make this a random topic? This way if can run multiple times the tests without things to interfere (if we have an existing deployment that does not get tear down at the end)

Maybe it makes sense to have something like:

`topic_password_rotation_{hash}`

I believe this would also allow us to have different database in MongoDB for each tests, if one wants to do some analytics aftwerward (and provided that MongoDB is not torn down)

Reply:
If we go down that route, worth pulling out the
```python
        key = "".join(random.choices(string.ascii_lowercase, k=4))
        generated_app_name = f"{role}-{key}"
```
And using that, per-app specific topic suffixes